doc/config.rst

```diff
@@ -297,3 +297,102 @@ recommended instead to use a *secure* local repository:
++-----------------------+------------------------------------------------------------------------------------------------------------------------------------+
+| ``happy``             | `<https://haskell-happy.readthedocs.io/en/latest/>`_                                                                               |
++-----------------------+------------------------------------------------------------------------------------------------------------------------------------+
+| ``haskell-suite``     | Haskell suite was abandoned long time ago.                                                                                         |
++-----------------------+------------------------------------------------------------------------------------------------------------------------------------+
+| ``haskell-suite-pkg`` | Haskell suite was abandoned long time ago.                                                                                         |
++-----------------------+------------------------------------------------------------------------------------------------------------------------------------+
+| ``hmake``             | Seems like hmake disappeared long time ago `<https://www.haskell.org/cabal/proposal-1.1/x756.html>`_                               |
++-----------------------+------------------------------------------------------------------------------------------------------------------------------------+
+| ``hpc``               | `<https://hackage.haskell.org/package/hpc>`_                                                                                       |
++-----------------------+------------------------------------------------------------------------------------------------------------------------------------+
+| ``hsc2hs``            | `<https://hackage.haskell.org/package/hsc2hs>`_                                                                                    |
++-----------------------+------------------------------------------------------------------------------------------------------------------------------------+
+| ``hscolour``          | `<https://hackage.haskell.org/package/hscolour>`_                                                                                  |
++-----------------------+------------------------------------------------------------------------------------------------------------------------------------+
+| ``jhc``               | `<http://repetae.net/computer/jhc/>`_                                                                                              |
++-----------------------+------------------------------------------------------------------------------------------------------------------------------------+
+| ``ld``                | Usually provided by GHC's ``"ld command"`` entry in ``ghc --info``.                                                                |
```

Comment:
Is it worth noting here that `ld` is only used by `ghci` and Template Haskell, and to build shared objects? Most things link via the C compiler, so that they get the C runtime and any global constructors/destructors required by that.

Reply:
I think I kind of mention that in the warning below. I.e that these programs are sometimes invoked via GHC and not directly so these options might not apply.

With the specific case of ld-options being applied also when calling GHC as `-optl`.